test/integration/extensions_test.py

```diff
@@ -0,0 +1,184 @@
+             'nulls': True, 'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'IS_FEMALE', 'type': 'BOOLEAN', 'sql_type': 'BOOLEAN', 'size': 1, 'scale': None, 'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'USER_RATING', 'type': 'DECIMAL', 'sql_type': 'DECIMAL(10,5)', 'size': 10, 'scale': 5,
+             'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'USER_SCORE', 'type': 'DOUBLE PRECISION', 'sql_type': 'DOUBLE', 'size': 64, 'scale': None,
+             'nulls': True, 'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'STATUS', 'type': 'VARCHAR', 'sql_type': 'VARCHAR(50) UTF8', 'size': 50, 'scale': None,
+             'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None}
+        ]
+        actual = connection.ext.get_sys_columns('users')
+        assert expected == actual
+
+    def test_get_sys_tables(self, connection):
+        expected = [
+            {'table_name': 'payments', 'table_schema': 'pyexasol_test', 'table_is_virtual': False,
+             'table_has_distribution_key': False, 'table_comment': None},
+            {'table_name': 'users', 'table_schema': 'pyexasol_test', 'table_is_virtual': False,
+             'table_has_distribution_key': False, 'table_comment': None}
+        ]
+        actual = connection.ext.get_sys_tables()
+        assert expected == actual
+
+    def test_get_sys_views(self, connection):
+        expected = []
+        actual = connection.ext.get_sys_views()
+        assert expected == actual
+
+    def test_get_sys_schemas(self, connection):
+        expected = [
+            {'schema_comment': None, 'schema_is_virtual': False, 'schema_name': 'pyexasol_test', 'schema_owner': 'sys'}
+        ]
+        actual = connection.ext.get_sys_schemas()
+        assert expected == actual
+
+    def test_get_disk_space(self, flush_statistics, connection):
+        expected = {
+            'free_size',
+            'measure_time',
+            'occupied_size',
+            'occupied_size_percent',
+            'total_size'
+        }
+        actual = set(connection.ext.get_disk_space_usage().keys())
+        assert expected == actual
+
+    def test_export_to_pandas_with_dtype(self, connection):
+        result = connection.ext.export_to_pandas_with_dtype("users")
+
+        expected = 10_000
+        actual = len(result)
+        assert actual == expected
+
+        expected = {
+            'user_id': 'int64',
+            'user_name': 'category',
+            'register_dt': 'datetime64[ns]',
+            'last_visit_ts': 'datetime64[ns]',
+            'is_female': 'category',
+            'user_rating': 'float64',
+            'user_score': 'float64',
+            'status': 'category'
+        }
+        actual = {column: result[column].dtype.name for column in expected}
```

Comment:
I assume in this test we are OK with the `result` having more columns than those in the `expected`.

Reply:
Yes it was intentional, thanks for mentioning it though, appreciate it.